### PR TITLE
[Ide+others] TooltipProvider.GetItem changed return type to "Task<TooltipItem>"

### PR DESCRIPTION
**Pull request by DavidKarlas**
Just review, don't merge I will merge manually because I have to push https://github.com/xamarin/md-addins/commit/d0b8299746c0ee41ca5d19b61f4f4a6ee1905e0b and F# which is using this API...

### CODE REVIEW

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.AnalysisCore/Gui/ResultTooltipProvider.cs [hunk 1]
```diff
@@ -38,7 +39,7 @@ public ResultTooltipProvider ()
 		{
 		}
 
-		public override TooltipItem GetItem (TextEditor editor, DocumentContext ctx, int offset)
+		public override Task<TooltipItem> GetItem (TextEditor editor, DocumentContext ctx, int offset)
```
Comment (slluis) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.AnalysisCore/Gui/ResultTooltipProvider.cs]: The return value of this method should be changed to Task.FromResult<TooltipItem> (null)

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Tooltips/LanguageItemTooltipProvider.cs [hunk 2]
```diff
@@ -67,7 +68,7 @@ public override string ToString ()
 		}
 
 		#region ITooltipProvider implementation 
-		public override TooltipItem GetItem (TextEditor editor, DocumentContext ctx, int offset)
+		public override Task<TooltipItem> GetItem (TextEditor editor, DocumentContext ctx, int offset)
 		{
 			if (ctx == null)
 				return null;
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Tooltips/LanguageItemTooltipProvider.cs]: Same as Lluis' comments from below, all these nulls should be `Task.FromResult(null)`.

**Comment by slluis** [comment]
Add a CancellationToken to GetItem. It can be used to cancel ResolveExpressionAsync, for example.

**Comment by DavidKarlas** [comment]
Updated to reflect suggested changes

**Comment by DavidKarlas** [comment]
merged manually as it also needed bumping of F# and md-addins

**Pull request closed without merging**